core/datasource/external/ethcall/engine.go

```diff
@@ -74,7 +75,7 @@ func NewEngine(log *logging.Logger, cfg Config, isValidator bool, client EthRead
 // Start starts the polling of the Ethereum bridges, listens to the events
 // they emit and forward it to the network.
 func (e *Engine) Start() {
-	if e.isValidator && e.client != nil {
```

Comment:
Code assumes the valid passed into eth engine will always be a pointer, I think this is OK?